routers/matchmaking.js

```diff
@@ -48,49 +48,126 @@ router.get("/:room_id/:subroom_id/all-instances", middleware.authenticateDevelop
     }
 });
 
+router.get("/my-instance-info", middleware.authenticateToken, async (req, res) => {
+    try {
+        /**
+         * @type {import('./ws/WebSocketServerV2').Connection}
+         */
+        const data = require('./ws/WebSocketServerV2').ws_connected_clients[req.user.id];
+        if (!data) return res.status(404).json({
+            code: "not_found",
+            message: "You are not online."
+        });
+
+        const instance = await GetInstanceById(data.roomId, data.instanceId);
+        if (!instance) return res.status(404).json({
+            code: "not_found",
+            message: "You are not in a room."
+        });
 
+        return res.status(200).json({
+            private: instance.MatchmakingMode == MatchmakingModes.Private,
+            instance_id: instance.InstanceId,
+            room_id: instance.RoomId,
+            subroom_id: instance.SubroomId,
+            age: instance.Age
+        });
+    } catch (ex) {
+        res.status(500).json({
+            code: "internal_error",
+            message: "An internal error occurred and we couldn't serve your request."
+        });
+        throw ex;
+    }
+});
 
-// yes this is literally just an enum
+/**
+ * @enum {number}
+ */
 const MatchmakingModes = {
     Public: 0,
     Unlisted: 1,
     Private: 2,
     Locked: 3
 };
 
-// YES, I'M AWARE
-// The fact that this system is entirely in-memory and therefore volatile is INTENTIONAL.
-// By design, when the API goes down, all users are removed from the game due to their websockets disconnecting.
-// This prevents failed API calls and cheating by faking a websocket connection.
-// HOWEVER - This also means that instances created by Photon will slowly be removed from the room list when all users are disconnected.
-// This means that the RoomSessions dictionary will be OUT OF DATE if it persists.
-// Therefore, it is in memory. This allows for both faster read/writes, better class implementation, more intellisense options,
-// and preventing the aforementioned persistence issues.
-
-// "Well, just make it so Photon Instances never expire, and you can persist as long as you like"
-// THIS IS A WASTE OF RESOURCES. It may also incur additional fees from Photon.
+/**
```

Comment:
A significant amount of incredibly useful documentation with IDE support using JSDoc.